### PR TITLE
connections and recontainment fixes

### PR DESCRIPTION
**Pull request by mariakleiner**

### CODE REVIEW

**Comment by mariakleiner** [comment]
one thing that i've noticed, but haven't fixed yet: when I recontain a Node, it is `invalidate`d, but without its state, to reproduce:
- Add `StaticText` node, in inspect set its value to "hello"
- Add a `Panel` node
- Drag `StaticText` into `Panel`: the object is being rebuilt, but `StaticText` atom is called with `undefined` (instead of "hello")
(it is fixed with refresh)...

I'll debug it some more later, but need to pause atm...

File: pkg/Library/Polymath/Atoms/WikipediaContent.js [hunk 1]
```diff
@@ -4,16 +4,20 @@ export const atom = (log, resolve) => ({
...
       return;
     }
-    const {title, html} = await service('PolymathService', 'QueryWikipedia', {query});
-    return {title, markup: html};
+    if (query) {
+      const {title, html} = await service('PolymathService', 'QueryWikipedia', {query});
+      return {title, markup: html};
+    } else {
+      return {
+        title: undefined,
```
Comment (sjmiles) [inline pkg/Library/Polymath/Atoms/WikipediaContent.js:17]: As a rule, we never set anything to `undefined`, only to `null`

**Review by sjmiles — APPROVED** [review]

**Comment by sjmiles** [comment]
> one thing that i've noticed, but haven't fixed yet: when I recontain a Node, it is `invalidate`d, but without its state, to reproduce:
> 
> * Add `StaticText` node, in inspect set its value to "hello"
> * Add a `Panel` node
> * Drag `StaticText` into `Panel`: the object is being rebuilt, but `StaticText` atom is called with `undefined` (instead of "hello")
>   (it is fixed with refresh)...
> 
> I'll debug it some more later, but need to pause atm...

Rebuilding the object seems like ridiculous overkill for something that only needs new rendering. I'll take a look.

File: pkg/Library/Polymath/Atoms/WikipediaContent.js (hunk 1 above)
Comment (mariakleiner) [reply pkg/Library/Polymath/Atoms/WikipediaContent.js]: I knew that, i don't know why i used `undefined`, changed to `null` now, thanks

**Comment by mariakleiner** [comment]
> > one thing that i've noticed, but haven't fixed yet: when I recontain a Node, it is `invalidate`d, but without its state, to reproduce:
> > 
> > * Add `StaticText` node, in inspect set its value to "hello"
> > * Add a `Panel` node
> > * Drag `StaticText` into `Panel`: the object is being rebuilt, but `StaticText` atom is called with `undefined` (instead of "hello")
> >   (it is fixed with refresh)...
> > 
> > I'll debug it some more later, but need to pause atm...
> 
> Rebuilding the object seems like ridiculous overkill for something that only needs new rendering. I'll take a look.

Thank you 🙏

**Pull request merged**